src/utils.js

```diff
@@ -42,8 +42,31 @@ export function printList(items, print, kind) {
         Tokens.scopeSpaceOrBreak(),
       ],
       print(item),
-      i === arr.length - 1 && Tokens.scopeEmptyOrComma(),
-      i === arr.length - 1 && Tokens.scopeEmptyOrBreak(),
+      i === arr.length - 1 && [
+        Tokens.scopeEmptyOrComma(),
+        Tokens.scopeEmptyOrBreak(),
+      ],
+    ]),
+    Tokens.scopeClose(),
+  ];
+}
+
+export function printJoin(items, print, separator) {
```

Comment:
Can we make separator a full token? Like scopeSeparator and `Tokens.scopeEmptyOrComma()` is passed in? I don't like using `scopeEmptyOr(...)` open ended like that.